src/EnergyPlus/ConvectionCoefficients.cc

```diff
@@ -1900,30 +2415,79 @@ void GetUserConvectionCoefficients(EnergyPlusData &state)
+                                             HcExt_ConvectionTypesMap,
+                                             ErrorsFound,
+                                             state.dataIPShortCut->cAlphaArgs(i),
+                                             state.dataIPShortCut->cAlphaArgs(i + 1),
+                                             state.dataIPShortCut->cAlphaFieldNames(i),
+                                             state.dataIPShortCut->cAlphaFieldNames(i + 1),
+                                             RoutineName,
+                                             CurrentModuleObject);
+
+        i += 2;
+        state.dataConvectionCoefficient->OutsideFaceAdaptiveConvectionAlgo.HNatVertWallEqNum =
+            SetOutsideAdaptiveConvectionAlgo(state,
+                                             HcExt_ConvectionTypesMap,
+                                             ErrorsFound,
+                                             state.dataIPShortCut->cAlphaArgs(i),
+                                             state.dataIPShortCut->cAlphaArgs(i + 1),
+                                             state.dataIPShortCut->cAlphaFieldNames(i),
+                                             state.dataIPShortCut->cAlphaFieldNames(i + 1),
+                                             RoutineName,
+                                             CurrentModuleObject);
+
+        i += 2;
+        state.dataConvectionCoefficient->OutsideFaceAdaptiveConvectionAlgo.HNatStableHorizEqNum =
+            SetOutsideAdaptiveConvectionAlgo(state,
+                                             HcExt_ConvectionTypesMap,
+                                             ErrorsFound,
+                                             state.dataIPShortCut->cAlphaArgs(i),
+                                             state.dataIPShortCut->cAlphaArgs(i + 1),
+                                             state.dataIPShortCut->cAlphaFieldNames(i),
+                                             state.dataIPShortCut->cAlphaFieldNames(i + 1),
+                                             RoutineName,
+                                             CurrentModuleObject);
+
+        i += 2;
+        state.dataConvectionCoefficient->OutsideFaceAdaptiveConvectionAlgo.HNatUnstableHorizEqNum =
+            SetOutsideAdaptiveConvectionAlgo(state,
+                                             HcExt_ConvectionTypesMap,
+                                             ErrorsFound,
+                                             state.dataIPShortCut->cAlphaArgs(i),
+                                             state.dataIPShortCut->cAlphaArgs(i + 1),
+                                             state.dataIPShortCut->cAlphaFieldNames(i),
+                                             state.dataIPShortCut->cAlphaFieldNames(i + 1),
+                                             RoutineName,
+                                             CurrentModuleObject);
+    }
```

Comment:
It's annoying, but honestly, I think it's a step in the right direction.  The other way was compact but very tough to follow.  Maybe someone can come up with a more clever solution.  I think the best way might be for the convection algorithm to hold a map instead of a bunch of individual EqNum's, then you could access the map by key, and just iterate over a list of keys.  I don't suggest we do that now, and I'm not suggesting that's the best way, but I think it could work.  Anyway, unrolling these is OK for now.